### DISCUSSION TITLE
Not understanding HuggingFace Model Upload

### DISCUSSION THREAD

**Discussion by InternalMegaT** [TheLastBen/fast-stable-diffusion#2432] [Q&A] [answered]
I'm trying to connect my base model on hugging face to my training sequence. For transfer learning, the issue is I'm keep getting this error:

```
HTTPError                                 Traceback (most recent call last)
<ipython-input-5-cfa16ae9bc50> in <cell line: 114>()
    115   if authe=="https://":
    116     textenc= f"{authe}huggingface.co/{Path_to_HuggingFace}/resolve/main/text_encoder/pytorch_model.bin"
--> 117     txtenc_size=urllib.request.urlopen(textenc).info().get('Content-Length', None)
    118   else:
    119     textenc= f"https://huggingface.co/{Path_to_HuggingFace}/resolve/main/text_encoder/pytorch_model.bin"

5 frames
/usr/lib/python3.10/urllib/request.py in http_error_default(self, req, fp, code, msg, hdrs)
    641 class HTTPDefaultErrorHandler(BaseHandler):
    642     def http_error_default(self, req, fp, code, msg, hdrs):
--> 643         raise HTTPError(req.full_url, code, msg, hdrs, fp)
    644 
    645 class HTTPRedirectHandler(BaseHandler):

HTTPError: HTTP Error 404: Not Found
```

How do I get the valid url for my huggingface model.
I used `https://huggingface.co/InternalMegaT/Internals_Pixel_Art_Diffusion/raw/main/Pixel%20Art%20Version%204.1%20Beta.ckpt` and `InternalMegaT/Internals_Pixel_Art_Diffusion/raw/main/Pixel%20Art%20Version%204.1%20Beta.ckpt`

**Comment by TheLastBen** [comment]
put the model link in the link input box, do a screenshot of the cell so I can understand the issue

**Reply by InternalMegaT** [reply]
> put the model link in the link input box, do a screenshot of the cell so I can understand the issue

I was putting the link in the Path_to_HuggingFace
What is the point of Path_to_HuggingFace if it doesn't allow paths?
I have now put it under link and works now.

**Reply by InternalMegaT** [reply]
Works is an understatement. I now get this error:
```
V2.1-768px Detected
Converting to Diffusers ...
Traceback (most recent call last):
  File "/content/convertodiff.py", line 1138, in <module>
    convert(args)
  File "/content/convertodiff.py", line 1083, in convert
    text_encoder, vae, unet = load_models_from_stable_diffusion_checkpoint(v2_model, args.model_to_load)
  File "/content/convertodiff.py", line 853, in load_models_from_stable_diffusion_checkpoint
    checkpoint = load_checkpoint_with_text_encoder_conversion(ckpt_path)
  File "/content/convertodiff.py", line 829, in load_checkpoint_with_text_encoder_conversion
    state_dict = checkpoint["state_dict"]
KeyError: 'state_dict'
rm: cannot remove 'stable-diffusion-custom': No such file or directory
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
Conversion error
```

**Reply by InternalMegaT** [reply]
![image.png](https://github.com/TheLastBen/fast-stable-diffusion/assets/75134841/5a577bb1-4b4b-4067-a812-0d198aeb94c6)

**Reply by SergioBoySV** [reply]
Hello @gagdsf2. I know it's been a while, but I too am having issues with selecting my own models to train with in the Colab journal. I am wondering why you included the `.../resolve/main/...` portion when you typed the link. When accessing Hugging Face models, is that a portion of the path that is supposed to be included in the link? 
<img width="1506" alt="Screenshot 2023-12-02 at 11 12 41 PM" src="https://github.com/TheLastBen/fast-stable-diffusion/assets/85202224/a486856c-0782-449f-9240-f30bad80fc5b">
I ask because I do not see that in InternalMegaT's repo for the model. Is that something I need to include across any other AI models I'd like to use from Hugging Face. Any help would be appreciated, thank you!

**Comment by TheLastBen** [comment, marked as answer]
fixed, run it again, it should work

**Reply by i-huzaifa-arshad** [reply]
Hi. I have returned to SD after a few months and I am using your colabs, but it is giving me the same "Not allowed" warning. Is this the end for running A1111 GUI on Free Colab? I tried encoding all possible keywords that could trigger this warning but still same warning. The same trick worked few months back. Please help. I also posted a question but no one answered.

**Reply by InternalMegaT** [reply]
> Hi. I have returned to SD after a few months and I am using your colabs, but it is giving me the same "Not allowed" warning. Is this the end for running A1111 GUI on Free Colab? I tried encoding all possible keywords that could trigger this warning but still same warning. The same trick worked few months back. Please help. I also posted a question but no one answered. 

You have to pay to use the colab Automatic1111, I recommend using alternatives posted all over the github. I can't find a reliable way to fix this.

**Reply by i-huzaifa-arshad** [reply]
i see. if you can atleast share the links of alternatives you are mentioning, that would be a huge favor.

**Comment by InternalMegaT** [comment]
> fixed, run it again, it should work

Oh this was a bug, sorry. I thought I was just using it wrong. It works now but I'm not sure how. The last revision was on July 22.

**Reply by TheLastBen** [reply]
the issue was in the conversion script, it's not in the notebook https://github.com/TheLastBen/fast-stable-diffusion/commit/b448b956fed5b867f91c921773a5d35da0c904f0
